flink/v1.16/flink/src/main/java/org/apache/iceberg/flink/FlinkWriteConf.java

```diff
@@ -153,16 +162,37 @@ public String orcCompressionStrategy() {
         .parse();
   }
 
-  public DistributionMode distributionMode() {
+  public DistributionMode distributionMode(List<Integer> equalityFieldIds) {
     String modeName =
         confParser
             .stringConf()
             .option(FlinkWriteOptions.DISTRIBUTION_MODE.key())
             .flinkConfig(FlinkWriteOptions.DISTRIBUTION_MODE)
             .tableProperty(TableProperties.WRITE_DISTRIBUTION_MODE)
-            .defaultValue(TableProperties.WRITE_DISTRIBUTION_MODE_NONE)
-            .parse();
-    return DistributionMode.fromName(modeName);
+            .parseOptional();
+
+    return modeName == null
+        ? defaultWriteDistributionMode(equalityFieldIds)
+        : DistributionMode.fromName(modeName);
+  }
+
+  private DistributionMode defaultWriteDistributionMode(List<Integer> equalityFieldIds) {
+    if (table.sortOrder().isSorted()) {
+      return RANGE;
+    } else if (table.spec().isPartitioned()) {
+      if (!equalityFieldIds.isEmpty()) {
+        PartitionSpec partitionSpec = table.spec();
+        for (PartitionField partitionField : partitionSpec.fields()) {
+          if (!equalityFieldIds.contains(partitionField.sourceId())) {
+            return NONE;
+          }
+        }
+      }
+
+      return HASH;
```

Comment:
is this logic correct? we shall only do HASH if equalityFieldIds list is not empty and it contains all the partition fields. right now, if the equalityFieldIds is empty, we will do HASH distribution.

Reply:
In the current default distribution policy, the HASH is returned only if the table is a partitioned, and equalityFieldIds is empty or equalityFieldIds contains all partitioned fields.
![image](https://user-images.githubusercontent.com/59213263/233534268-45c1c8a8-8d2f-4395-915b-a03271815e8e.png)